### ISSUE TITLE
Redelegated Voting Power to be assigned immediately 

### ISSUE THREAD

**Issue by rigelrozanski** [cosmos/cosmos-economics#3]
There has been some discussions as to whether or not it is safe to allow for power from the re-delegated atoms to be assigned immediately even while the unbonding period from the original bond is taking place - under this scenario the re-delegated atoms would be slashable for the actions of both the old delegator and the new delegator.. Possible safety implications? 

Possible process flow discussed by Jae and I - when re-delegating the hold account atoms are automatically moved from the old validator to the new validator. Within the new delegator - the atoms are treated as a regular delegation. Within the old delegator, it's atoms have already been removed from the an item is added to a list of it's "rebonding" atoms - if the old validator becomes slashed the slashable value from the "rebonding" atoms are assigned as debt to the delegator addresses. If a delegator address has dept - the only operation they are permitted to perform is an unbond operation, additionally they cannot vote, or accrue and new fees from the time the debt was added.... other incentives?

**Comment by rigelrozanski** [comment]
from @ethanfrey 
I disagree with this point strongly.

A group collectively bonds 40% of total voting power to 10 validators.
Block H they all rebond to innocent validators.

A forked Block H+20 is presented to an unsuspecting light client signed by 70% of the stake,
as defined by the validator set in block H (same as what the light client know, from block H-20).

Another one is also signed by 70% (both 30% normal, plus 40% double signers).

At this point we can have 1/3 < stake < 2/3 fork the chain without being at risk for their atoms, which were no longer at risk for the actions of those 10 malicious validators.

**Comment by rigelrozanski** [comment]
@jaekwon discussed why this is not an issue earlier today - It would be good to have this written down. I think the gist is, validators are also punished for publishing on malicious chains. As soon as a malicious fork is detected a network halt and block reorg and slashing is to take place.

The decision is that instant rebond is acceptable. But I still do believe that we need a better documentation of details pertaining to Frey's concern

**Comment by adrianbrink** [comment]
@rigelrozanski Can you document this decision the way we documented #11 . Otherwise it is hard to understand the pros and cons and why a certain decision was made.

**Comment by rigelrozanski** [comment]
Based on further contemplation with Jae and Adrian: We determined that the safest and simplest mechanism to implement which would incentivize switching delegators was to reward provisions and fees based during the unbonding period, and base the calculation of global fee distribution and provision distribution on all slashable validator tokens - however voting power is still only based on validator tokens which have not yet begun the unbonding process… This is reasonable because the as soon as somebody begins unbonding that validator now has less power to mount a malicious attack on the system with, while simultaneously if they do screw up somehow, there is more on the line to be lost. This idea stretches back to some of the earlier discussions we had were we decided it would be safe to base provisions not on the voting power but on the slashable supply.

**Comment by sunnya97** [comment]
But wasn't the issue with this that there is then no incentive for a delegator to not just always remain in the unbonding phase.

**Comment by rigelrozanski** [comment]
As per our discussion, I think it would be valueable to reinvestigate the safety of the original idea post launch - def not a prelaunch item based on solution involving rebonding as just outlined. Also there is no incentive to always be in the unbonding phase because while unbonding you are still slashible for actions of the validator even if they didn't include your voting power

**Comment by sunnya97** [comment]
If we don't have immediate rebonding, what do we do with a validator who was kicked out of the top 100?  Do they start unbonding to liquidity and thus not get any rewards that entire unbonding period time?  Or do the unbond-to-redelegate?  In which case, who are they getting re-delegeated to?

**Comment by rigelrozanski** [comment]
> what do we do with a validator who was kicked out of the top 100? Do they start unbonding to liquidity and thus not get any rewards that entire unbonding period time?

This is correct under the assumption that we move forward with no provisions or fee allocation for those unbonding to liquidity (does not apply to rebonding).
> Or do the unbond-to-redelegate? In which case, who are they getting re-delegeated to?

If a delegator's bond is being unbonded with the whole validator, they may during this validator unbonding process start re-delegating to a new validator... during the remainder of the unbonding period from the re-delegating process if the old validator was to start validating again (let's say they were kicked for availability, but then they came back online) and start collecting fees/provisions then the re-delegating delegator would by default also be collecting these rewards during the remainder of their unbonding period which is being applied as they redelegate. If the old validator was to not come back online then the re-delegating delegator would not be able to collect fees during the unbonding period or re-delegation...

It's a bit of a mouthful, let me know if you want more clarity. We could of course structure this so that unbonding to liquidity would still afford you fees/provisions, in which case its more favourable as a delegator and you'd be collecting during all the above scenarios, but I don't think we want that.

**Comment by sunnya97** [comment]
It makes sense, but I then this creates high disincentive against delegating to the bottom validators due to risk that they might be pushed out.

**Comment by rigelrozanski** [comment]
Hmm - so if a validator at the bottom of the ring is pushed out they would be required to begin unbonding to liquidity which means that they are no longer collecting rewards... Okay, so I guess always providing rewards even when you're unbonding would be a solution to this... I still think it's going to be hard to convince people to bond at the very bottom because even though you could re-delegate if your validator gets kicked and not lose any value, you will still have to be on your toes because this should happen with some reasonable frequency to the bottom most few delegators... Any further thoughts? I'm thinking that if we are going to provide rewards during unbonding to liquidity we will need to provide them under all scenarios and for all actors... We would need to compensate this with a bit more slashing for liveliness faults when you're getting kicked - but that should be okay.

**Comment by sunnya97** [comment]
I think the initial proposal of allowing immediate redelegation solves this.

**Comment by rigelrozanski** [comment]
um yeah well this also resolves this! I don't think it makes any sense to include instant re-delegation until there is a more thorough evaluation and specing as to its mechanics and implications... I do invite you to begin this process but it doesn't seem small, and I think we are going to find there is a wider set of complex interactions we will/would need to build out for... again all for understanding this it's just really not MVP material for as far as I can tell (open to being proven wrong with a comprehensive spec!)

**Comment by sunnya97** [comment]
Well we can't provide rewards during unbonding to liquidity because of the fact then there's no incentive to not just remain forever unbonding.  So for MVP, I guess if you get pushed out of the bottom, you just take the hit and be stuck without rewards for the unbonding period?

**Comment by sunnya97** [comment]
Update:  I just realized, we should ever give rewards unless voting power is also added, even if unbonding to redelegate, because as a delegators, there is still an incentive to remain in the unbonding to redelegation positition at all times.  Yes, your coins are still at stake, but only for the times when your voting power was being used by that validator.  Because there is a risk that a validator might double sign at any point, you're incentivized to always unbond to redelegate as soon as you bond to a new validator.  That way you are only liable for the validator's actions for that short period (like 1 block) that you were actively bonded to them.

**Comment by rigelrozanski** [comment]
> Oh what if during bonding to redelegate, you get the rewards from the new validator, not the previous one? That way, partway through unbonding period to liquidity, you can switch to unbonding to redelegation, and start earning rewards from that point?

Oh cool idea, don't see anything wrong with this. I need to think about implementation 

Also to your second comment I came to this realization a while ago which is why we need to be slashable to the old validators CURRENT actions not just while your voting power was contributing this alleviates the problem you have outlined

**Comment by sunnya97** [comment]
Wait, but then that creates the situation where what if at the end of your unbonding period, the validator does something malicious, you needs to be slashed.  But by the time that that proof is submitted, you've already moved to a new validator.

**Comment by rigelrozanski** [comment]
If there is slashable event instigated during their unbonding period which was discovered after completion of the re-bond, the rebonding delegator does not need to be slashed as their voting power would have not contributed to that event - so we don't need to guarantee their slashability. Doesn't matter because the only reason you are being slashed for events where your voting power is not contributing to them is to prevent the situation you outlined which incentivizes folks to be constantly rebonding. It's more like a bonus slash as the protocol doesn't need to guarantee slashing those atoms for safety reasons
